crates/matrix-sdk-crypto/src/store/crypto_store_wrapper.rs

```diff
@@ -109,10 +120,71 @@ impl CryptoStoreWrapper {
             // Mapping the devices and user identities from the read-only variant to one's
             // that contain side-effects requires our own identity. This is
             // guaranteed to be up-to-date since we just persisted it.
-            let own_identity =
+            let maybe_own_identity =
                 self.store.get_user_identity(&self.user_id).await?.and_then(|i| i.into_own());
 
-            let _ = self.identities_broadcaster.send((own_identity, identities, devices));
+            // If our identity was not verified before the change and is now, that means
+            // this could impact the verification chain of other known
+            // identities.
+            if let Some(own_identity_after) = maybe_own_identity.as_ref() {
+                // Only do this if our identity is passing from not verified to verified,
+                // the previously_verified can only change in that case.
+                if !own_identity_was_verified_before_change && own_identity_after.is_verified() {
+                    debug!("Own identity is now verified, check all known identities for verification status changes");
+                    // We need to review all the other identities to see if they are verified now
+                    // and mark them as such
+                    self.check_all_identities_and_update_was_previousy_verified_flag_if_needed(
```

Comment:
Thank god for `typos`.
```suggestion
                    self.check_all_identities_and_update_was_previously_verified_flag_if_needed(
```